### PR TITLE
[JetBrains] force upgrade IntelliJ IDEA to `2024.2`

### PR DESCRIPTION
**Pull request by mustard-mh**
## Related Issue(s)

Fixes ENT-651

## How to test

- Wait for last GHA to finished
- Open Preview env, create workspace with stable IntelliJ IDEA
- It should be able to launch and connect using Gateway + IntelliJ IDEA
- GHA's `integration tests` should be passed

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - hw-ij-upgrade</li>
	<li><b>🔗 URL</b> - <a href="https://hw-ij-upgrade.preview.gitpod-dev.com/workspaces" target="_blank">hw-ij-upgrade.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - hw-ij-upgrade-gha.28077</li>
	<li><b>🗒️ Logs</b> - <a href="https://console.cloud.google.com/logs/query;query=jsonPayload.kubernetes.host%3D%22preview-hw-ij-upgrade%22%0A%0A--%20Filter%20on%20service:%0A--%20jsonPayload.serviceContext.service%3D%22ws-manager-mk2%22%0A;duration=P1D?project=gitpod-dev-preview" target="_blank">GCP Logs Explorer</a></li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [x] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [x] /werft preemptible
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
- [x] with-integration-tests=jetbrains
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

**Comment by geropl** [comment]
The image build failed with: `no space left on device` :see_no_evil: 

@mustard-mh I will re-create with a bigger VM

File: dev/preview/workflow/preview/patch-ide-configmap.js [hunk 1]
```diff
@@ -24,6 +29,18 @@ for (let ide in json.ideOptions.options) {
...
+    if (
+        ["intellij"].includes(
+            ide,
+        )
+    ) {
+        json.ideOptions.options[ide].pluginImage = replaceImage2(json.ideOptions.options[ide].pluginImage)
+        json.ideOptions.options[ide].imageLayers = json.ideOptions.options[ide].imageLayers.map(replaceImage2)
+    }
+
+    json.ideOptions.options[]
```
Comment (geropl) [inline dev/preview/workflow/preview/patch-ide-configmap.js:43]: This line is to be removed.

**Comment by geropl** [comment]
Still seeing this, even on a fresh workspace, after hitting "Don't show again" on a previous one.
Something we need to solve before merging?

image

**Review by geropl — APPROVED** [review]
Intellij itself works nicely as far as I tested the basics :heavy_check_mark:

**Comment by mustard-mh** [comment]
One last smoke test. Waiting for integration tests

image

**Comment by mustard-mh** [comment]
There's a problem for integration tests (it doesn't load Gitpod Gateway plugin). So we need to ignore integration tests now

image

Manually tested with stable WebStorm and GoLand
image

**Pull request merged**